test/bh/applyNext-test.js

```diff
@@ -0,0 +1,138 @@
+var fixtures = require('../fixtures')('bemhtml');
+var compile = fixtures.compile;
+
+describe('.applyNext()', function() {
+  it('should apply templates for new mod', function() {
+    compile(function() {
+      block('button').def()(function() {
+        this.mods.type = 'span';
+        return applyNext();
+      });
+      block('button').mod('type', 'span').tag()('span');
+    })
+      .apply({ block: 'button' })
+      .should.equal(
+        '<span class="button button_type_span"></span>'
+      );
+  });
+
+  it('should apply base matcher for element', function() {
+    compile(function() {
+      block('button').elem('control').def()(function() {
+        this.elemMods.type = 'span';
+        return applyNext();
+      });
+      block('button').elem('control').elemMod('type', 'span').tag()('span');
+    })
+      .apply({ block: 'button', elem: 'control', elemMods: { disabled: true } })
+      .should.equal(
+        '<span class="button__control button__control_disabled' +
+        ' button__control_type_span"></span>'
+      );
+  });
+
+  it('should apply base matcher for content', function() {
+    compile(function() {
+      block('button').content()(function() {
+        return [
+          { elem: 'base-before' },
+          applyNext(),
+          { elem: 'base-after' }
+        ];
+      });
+      block('button').content()(function() {
+        return [
+          { elem: 'before' },
+          applyNext(),
+          { elem: 'after' }
+        ];
+      });
+    })
+      .apply({ block: 'button', content: 'Hello' })
+      .should.equal(
+        '<div class="button">' +
+          '<div class="button__before"></div>' +
+          '<div class="button__base-before"></div>' +
+          'Hello' +
+          '<div class="button__base-after"></div>' +
+          '<div class="button__after"></div>' +
+        '</div>'
+      );
+  });
+
+  xit('should apply base matcher while wrapping', function() {
+    compile(function() {
+      block('button').content()(function() {
+        return [
+          { elem: 'base-before' },
+          applyNext(),
+          { elem: 'base-after' }
+        ];
+      });
+      block('button').content()(function() {
+        var res = applyNext();
+        return [
+          { elem: 'before' },
+          this.ctx,
```

Comment:
IMHO Nop. [We already have enough](https://github.com/bem/bem-xjst/blob/master/test/bemtree/runtime-test.js#L8-L100).